railties/lib/rails/commands/commands_tasks.rb

```diff
@@ -26,20 +30,23 @@ class CommandsTasks # :nodoc:
  runner       Run a piece of code in the application environment (short-cut alias: "r")
 
 All commands can be run with -h (or --help) for more information.
+
 EOT
 
-    COMMAND_WHITELIST = %w(plugin generate destroy console server dbconsole runner new version help test)
+    COMMAND_WHITELIST = %w(plugin generate destroy console server dbconsole runner new version help test rake)
 
-    def initialize(argv)
+    def initialize(argv = ARGV)
       @argv = argv
     end
 
     def run_command!(command)
       command = parse_command(command)
+
       if COMMAND_WHITELIST.include?(command)
         send(command)
       else
-        write_error_message(command)
+        ARGV.unshift(command)
+        send(:rake)
```

Comment:
Renaming result variable of `parse_command` won't help here, as the ARGV.shift is done at https://github.com/rails/rails/blob/master/railties/lib/rails/commands.rb#L13 and we need to put it back/unshift it for rake to take over, otherwise rake won't be able to find the right task to run and defaults to the `rake test`.